### DISCUSSION TITLE
Translation: Assessment of the difficulty in porting CPU architecture for firebird

### DISCUSSION THREAD

**Discussion by wangyuliu** [FirebirdSQL/firebird#7792] [General]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **firebird** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be high, with a moderate amount of code related to the CPU architecture in the project.  Is this assessment accurate?Do you have any opinions on personnel allocation and consumption time？ I look forward to your help and response.

**Comment by aafemt** [comment]
Most likely you tool get fooled by ifdef-ed code. Firebird uses non-standard macros to determine current platform in platform-depended code. Besides, porting cost depends mostly not on CPU but on OS.
